AuthenticatorBridgeKit/CodableModelData.swift

```diff
@@ -0,0 +1,40 @@
+import CoreData
+import OSLog
+
+/// A protocol for a `NSManagedObject` which persists a data model as JSON encoded data. The model
+/// can be set via the `model` property which encodes the model to the data property, which should
+/// be a `@NSManaged` property of the `NSManagedObject`. When the managed object is populated from
+/// the database, the `model` property can be read to decode the data.
+///
+protocol CodableModelData: AnyObject, NSManagedObject {
+    associatedtype Model: Codable
+
+    /// A `@NSManaged` property of the manage object for storing the encoded model as data.
+    var modelData: Data? { get set }
+}
+
+extension CodableModelData {
+    /// Encodes or decodes the model to/from the data instance.
+    var model: Model? {
+        get {
+            guard let modelData else { return nil }
+            do {
+                return try JSONDecoder().decode(Model.self, from: modelData)
+            } catch {
+                Logger.bridgeKit.error("Error decoding \(String(describing: Model.self)): \(error)")
```

Comment:
🤔 Should we log to crashlytics (if possible, i.e. use `ErrorReporter`) if an error happens here? Or directly not catch it and let the caller handle it? I mean this would be pretty hidden if it happens to an end user.  Maybe I'm being overcautious here but wanted to raise it just in case.

Reply:
In the main PM app's `CodableModelData` it just logs to Logger.application - so I was following that as a guide. I think that's probably the best choice though:
- This is coding/decoding the object that our code put in there (as opposed to say, something from a server). So it's pretty unlikely it would be of an incorrect format. And if it was, it's pretty unclear how we'd recover.
- Since these are in getter/setters it is a little awkward to throw the error out of this context. Instead, we're just returning `nil` and requiring that the property be Optional. So any error handling is likely going to be done where we unwrap that and use the value in the caller.

Reply:
Makes sense thanks!